### ISSUE TITLE
Unescape/replace `\(` and `\)` before passing to link-check

### ISSUE THREAD

**Issue by nschonni** [tcort/markdown-link-check#37]
Example links in markdown to old MSDN docs like https://github.com/MicrosoftDocs/visualstudio-docs/blame/e86fee5a5a0ec1a573b9cd153ee1cb6bc91b64ba/scripting-docs/javascript/reference/arraybuffer-object.md#L63

```markdown
from an [XMLHttpRequest](http://msdn.microsoft.com/library/ie/ms535874\(v=vs.85\).aspx).
```
These appear to be a valid way of escape braces in a markdown link, but appears like so when running through the validator
```
[✖] http://msdn.microsoft.com/library/ie/ms535874\(v=vs.85\
```

**Comment by nschonni** [comment]
Actually, I think https://github.com/tcort/markdown-link-extractor is the right repo for this

**Comment by nschonni** [comment]
Reopening since the marked library got a bump and I've PR'd that to the markdown-link-extractor repo

**Comment by tcort** [comment]
I've published a new version, v3.5.2, with updated dependencies which properly handles parentheses in URLs.

**Comment by nschonni** [comment]
Thanks, verified it works now :+1:
